server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ACVRDownloadByCounty.java

```diff
@@ -60,18 +65,58 @@ public String endpointName() {
   public String endpoint(final Request the_request, final Response the_response) {
     final String[] counties = the_request.params(":counties").split(",");
     final Set<String> county_set = new HashSet<String>(Arrays.asList(counties));
+    String result = "";
+    int status = HttpStatus.OK_200;
+    
     try {
       final OutputStream os = SparkHelper.getRaw(the_response).getOutputStream();
       final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
-      
-      Main.GSON.toJson(CastVoteRecord.getMatching(county_set, 
-                                                  RecordType.AUDITOR_ENTERED),
-                       bw);
-      bw.flush();
-      return "";
+      final Set<CastVoteRecord> matches = getMatching(county_set);
+      if (matches == null) {
+        result = "Unable to fetch records from database";
+        status = HttpStatus.INTERNAL_SERVER_ERROR_500;
+      } else {
+        Main.GSON.toJson(matches, bw);
+        bw.flush();
+        result =  "";
+      }
     } catch (final IOException e) {
-      the_response.status(HttpStatus.INTERNAL_SERVER_ERROR_500);
-      return "Unable to stream response.";
+      status = HttpStatus.INTERNAL_SERVER_ERROR_500;
+      result = "Unable to stream response.";
+    }
+    
+    the_response.status(status);
+    return result;
+  }
+  
+  
+  /**
+   * Returns the set of ACVRs matching the specified county IDs.
+   * 
+   * @param the_county_ids The set of county IDs.
+   * @return the ACVRs matching the specified set of county IDs.
+   */
+  private Set<CastVoteRecord> getMatching(final Set<String> the_county_ids) {
+    Set<CastVoteRecord> result = null;
+
+    try {
+      final Set<CastVoteRecord> query_result = new HashSet<>();
+      Persistence.beginTransaction();
+      final Session s = Persistence.currentSession();
+      for (final String county_id : the_county_ids) {
+        final Query<CastVoteRecord> query = 
+            s.createQuery("from CastVoteRecord where county_id = '" + 
```

Comment:
Queries aren't likely to stay created this way anyway. I was intending to convert them to type safe JPA queries rather than writing any SQL by hand, I just hadn't gotten to that yet.